src/include/sof/math/auditory.h

```diff
@@ -0,0 +1,114 @@
+/* SPDX-License-Identifier: BSD-3-Clause
+ *
+ * Copyright(c) 2022 Intel Corporation. All rights reserved.
+ *
+ * Author: Seppo Ingalsuo <seppo.ingalsuo@linux.intel.com>
+ */
+
+/* Psychoacoustics related functions */
+
+#ifndef __SOF_MATH_AUDITORY_H__
+#define __SOF_MATH_AUDITORY_H__
+
+#include <sof/math/fft.h>
+#include <sof/math/log.h>
+#include <stdint.h>
+
+#define AUDITORY_EPS_Q31	1 /* Smallest nonzero Q1.31 value */
+#define AUDITORY_LOG_2P31_Q26	Q_CONVERT_FLOAT(21.4875625974, 26) /* log(2^31) */
+#define AUDITORY_LOG2_2P25_Q16	Q_CONVERT_FLOAT(25.0, 16) /* log2(2^25) */
+
+enum psy_mel_log_scale {
+	MEL_LOG = 0,
+	MEL_LOG10,
+	MEL_DB,
+};
+
+/* Struct to define Mel filterback calculation. The filterbank data is compressed into
+ * a single vector from normal (half_fft_bins, mel_bins) size by storing only non-zero
+ * weights values.
+ *
+ * Triangles data Number triangles is mel_bins
+ * then for each triangle:
+ *	index to next triangle 0, start fft_bin 1, length of this triangle 2,
+ *	triangle weight values 3..N
+ */
+struct psy_mel_filterbank {
+	int32_t log_mult; /**< Out, QX.Y scale for log, log10, or dB format */
+	int32_t scale; /**< Out, Scale Q16.16 used for Q1.15 triangles */
+	int32_t scale_inv; /**< Out, Inverse scale Q16.16 applied to triangles data */
```

Comment:
Fields scale and scale_inv can be converted to local variables. The correction for Mel triangles gain is done with scale_log2 now.